### PR TITLE
ENDOC-711 CM-env-vars

### PR DESCRIPTION
**Pull request by jyunmitch**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -23,19 +23,30 @@ EntandoDeBundles served in the cluster can be installed in an EntandoApp using t
...
-| ENTANDO_BUNDLE_TYPE                                           | The bundle type that should be handled by this service. It can be `git` or `npm`. The default value is `git`. |
+| Group | Name | Value [default] | Description |
+| :---   | :--- | :--- |:--- |
+| General | ENTANDO_URL                 |  | The URL to access the Entando App instance                            |
+|     | ENTANDO_APP_NAMESPACE                                        | [test-namespace] | The kubernetes namespace where the entando app is running |
+|     | ENTANDO_APP_NAME                                             | [test-entando] | The entando app name that this service is in |
+|    | ENTANDO_ECR_DEAPP_REQUEST_RETRIES    | [3] |  Number of times the CM retries the component create/update before giving up     |
+|    | ENTANDO_ECR_DEAPP_REQUEST_BACKOFF       | [5] |  Seconds to wait before the next attempt is executed     |
+|    | ENTANDO_ECR_POSTINIT        |  |  Configuration of the postinit process      |
+|    | ENTANDO_CONTAINER_REGISTRY_CREDENTIALS   | [null] | Credentials for each container registry    |
```
Comment (nshaw) [inline README.md]: Is null really the default?

File: README.md [hunk 2]
```diff
@@ -23,19 +23,30 @@ EntandoDeBundles served in the cluster can be installed in an EntandoApp using t
...
+|    | MANAGEMENT_ENDPOINTS_WEB_EXPOSURE_INCLUDE     | [not enabled] |  To enable component-manager /actuator/info     |
```
Comment (nshaw) [inline README.md]: [not enabled] doesn't seem like a default. I would expect this to be true, [false]

File: README.md [hunk 3]
```diff
@@ -23,19 +23,30 @@ EntandoDeBundles served in the cluster can be installed in an EntandoApp using t
...
+|     |SPRING_SECURITY_OAUTH2_CLIENT_REGISTRATION_OIDC_CLIENT-ID   |  | The client id for the service                                                                                 |
+|     |SPRING_SECURITY_OAUTH2_CLIENT_REGISTRATION_OIDC_CLIENT-SECRET| | The client secret         |
+| Database | DB_VENDOR             | [postgres] | Which database will be used     |
+|    | DB_HOST      | [localhost] | Database host |
+|    | DB_PORT  | [5432] | Database port |
+|    | DB_NAME    | [digital_exchange] | Database name |
+|    | DB_OPTIONS           | [useSSL=false] | Database options |
+|    | DB_USER       | [admin] | Database user |
+|    | DB_PASS        | [admin] | Database password |
+| Bundle Related | ENTANDO_BUNDLE_TAGS_TYPES       | dev, [prod] | To generate EntandoDeBundle CRs using tags to select for dev, prod, or both |
```
Comment (nshaw) [inline README.md]: I think this can be a comma-separated list so might be worth mentioning that in the description

File: README.md [hunk 4]
```diff
@@ -23,19 +23,30 @@ EntandoDeBundles served in the cluster can be installed in an EntandoApp using t
...
+|     | ENTANDO_BUNDLE_TYPE           | npm, [git] | The bundle type that should be handled by this service |
+|    | ENTANDO_BUNDLE_DOWNLOAD_TIMEOUT    | [300] | Download timeout in seconds   |
+|    | ENTANDO_BUNDLE_DOWNLOAD_RETRIES   | [3] |  Max download attempts         |
+|    | ENTANDO_BUNDLE_DECOMPRESS_TIMEOUT | [600] |  Decompress timeout in seconds     |
+|    |          | [] |       |
```
Comment (nshaw) [inline README.md]: Extra line to remove here.

File: README.md [hunk 5]
```diff
@@ -23,19 +23,30 @@ EntandoDeBundles served in the cluster can be installed in an EntandoApp using t
...
+|     |SPRING_SECURITY_OAUTH2_CLIENT_REGISTRATION_OIDC_CLIENT-SECRET| | The client secret         |
+| Database | DB_VENDOR             | [postgres] | Which database will be used     |
+|    | DB_HOST      | [localhost] | Database host |
+|    | DB_PORT  | [5432] | Database port |
+|    | DB_NAME    | [digital_exchange] | Database name |
+|    | DB_OPTIONS           | [useSSL=false] | Database options |
+|    | DB_USER       | [admin] | Database user |
+|    | DB_PASS        | [admin] | Database password |
+| Bundle Related | ENTANDO_BUNDLE_TAGS_TYPES       | dev, [prod] | To generate EntandoDeBundle CRs using tags to select for dev, prod, or both |
+|     | ENTANDO_BUNDLE_TYPE           | npm, [git] | The bundle type that should be handled by this service |
```
Comment (nshaw) [inline README.md]: 1. We removed npm support a long time ago. I think we should remove this from the list. 
2. Is this var even used now?  Even the git type is deprecated now so I'm not sure what this one does anymore.

**Review by nshaw — COMMENTED** [review]
@firegloves @entgigi Might need your input on a couple of my questions.

File: README.md (hunk 1 above)
Comment (entgigi) [reply README.md]: `@Value("${entando.container.registry.credentials:#{null}}")` Yes, it is. You could write `blank`, I think it is better than null (and we check for not blank in that property).

File: README.md (hunk 2 above)
Comment (entgigi) [reply README.md]: the endpoint `/actuator/info` is not enabled by default, to enable it you should use the following spring-boot params or env var: `management.endpoints.web.exposure.include=health,info` 
I think that by default management.endpoints.web.exposure.include is `health` and the env var is empty
I suggest referring to the spring boot docs https://docs.spring.io/spring-boot/docs/2.5.x/reference/html/actuator.html

File: README.md (hunk 1 above)
Comment (jyunmitch) [reply README.md]: changed to blank

File: README.md (hunk 5 above)
Comment (jyunmitch) [reply README.md]: removed from list

File: README.md [hunk 6]
```diff
@@ -23,19 +23,29 @@ EntandoDeBundles served in the cluster can be installed in an EntandoApp using t
...
+|     |SPRING_SECURITY_OAUTH2_CLIENT_REGISTRATION_OIDC_CLIENT-ID   |  | The client id for the service                                                                                 |
+|     |SPRING_SECURITY_OAUTH2_CLIENT_REGISTRATION_OIDC_CLIENT-SECRET| | The client secret         |
+| Database | DB_VENDOR             | [postgres] | Which database will be used     |
+|    | DB_HOST      | [localhost] | Database host |
+|    | DB_PORT  | [5432] | Database port |
+|    | DB_NAME    | [digital_exchange] | Database name |
+|    | DB_OPTIONS           | [useSSL=false] | Database options |
+|    | DB_USER       | [admin] | Database user |
+|    | DB_PASS        | [admin] | Database password |
+| Bundle Related | ENTANDO_BUNDLE_TAGS_TYPES       | dev, [prod] | To generate EntandoDeBundle CRs using tags to select for dev, prod, or both as a comma delimited list|
```
Comment (firegloves) [inline README.md:46]: Should we change `[prod]` into `prod`?

**Review by firegloves — CHANGES_REQUESTED** [review]

**Review by firegloves — APPROVED** [review]
LGTM

**Pull request merged**